Documenting multithreading enhancement

jaymedina opened the pull request:
This PR updates the Astrocut documentation to reflect the work done in PR https://github.com/spacetelescope/astrocut/pull/84

These are the new changes:

<img width="871" alt="image" src="https://github.com/spacetelescope/astrocut/assets/32107699/584d2ef3-1ec1-4179-9c0e-e4c38fdc0158">

falkben commented on docs/astrocut/index.rst:
```diff
@@ -219,7 +219,7 @@ Making cutout target pixel files
 
 To make a cutout, you must already have an image cube to cut out from.
-Assuming that that step has been completed, you simply give the central
+Assuming that you have a TESS cube file stored locally, you can simply give the central
```
I know not your change, but can we remove all instances of the word "simply" from our docs?

falkben commented on docs/astrocut/index.rst:
```diff
@@ -256,7 +256,60 @@ The cutout target pixel file format is decribed `here <file_formats.html#target-
+                >>> cutout_size = 30
+                >>> cube_file = "s3://stpubdata/tess/public/mast/tess-s0038-2-2-cube.fits"
+
+                >>> cut_factory.cube_cut(cube_file, coordinates=coord, cutout_size=cutout_size, verbose=True, threads="auto") #doctest: +SKIP
```
Can we put some line breaks here so the `threads` option doesn't scroll off the side and is more easily visible?

falkben reviewed (changes requested):
This looks really good, thanks for adding this.

scfleming commented:
"To use multiprocessing, set the `threads` ... argument to an integer greater than 1".  I assume this is actully the number of threads you want to use, as written it makes it sounds like any interger greater than 1 just turns on the feature.  Perhaps we can say "To use multiprocessing, set the `threads` .... argument to the number of threads you want to use.  Setting it to a value of 1 disables the use of threads(? is that right?).  Setting it to `auto`..."

scfleming commented:
Towards the end it says "By default multithreading is disabled by default, but you can also disable it by setting `threads` <= 1".  At first I misread this and thought it was backwards but noticed it says "disable it" and not "enable it".  If the default is to not enable it, we could just indicate how to enable it (by setting threads > 1) and avoid discussing alternative ways of disabling it to avoid confusing users?  The behavior itself is good (that's naturally what someone should expect this to do if given a # of threads of 1 or less) but I don't feel like it's instructive for users to know that in the documentation.

jaymedina commented:
> If the default is to not enable it, we could just indicate how to enable it (by setting threads > 1) and avoid discussing alternative ways of disabling it to avoid confusing users? The behavior itself is good (that's naturally what someone should expect this to do if given a # of threads of 1 or less) but I don't feel like it's instructive for users to know that in the documentation.

Yep, that's fair, I've removed the comment about how to disable multiprocessing, and just mention that multiprocessing is disabled by default, and updated the second example (without multiprocessing) accordingly.

falkben commented on docs/astrocut/index.rst:
```diff
@@ -256,7 +256,62 @@ The cutout target pixel file format is decribed `here <file_formats.html#target-
                 2  APERTURE      1 ImageHDU        45   (5, 5)   float64  
 
-  
+You can also create cutout target pixel files out of TESS cube files stored on MAST's AWS open data bucket.
```
Should we add a section header above this?

jaymedina replied:
I made it a subsection so that it still falls under the `TESS Full-Frame Image Cutouts` section. And the `Multithreading` subsection is just below it. Here's what it looks like now: 

<img width="732" alt="image" src="https://github.com/spacetelescope/astrocut/assets/32107699/80e1705f-0ae0-4868-84aa-68eddf6e569b">

falkben replied:
Can we call this multithreading instead? since multiprocessing and multithreading are very different things, people might get confused if we use the other term.

jaymedina replied:
Sure thing

hmlewis-astro reviewed (changes requested):
Just a few places to go through and change multiprocessing -> multithreading, but otherwise, looks good!
